airflow/config_templates/default_airflow.cfg

```diff
@@ -325,6 +325,11 @@ scheduler_zombie_task_threshold = 300
 # DAG definition (catchup)
 catchup_by_default = True
 
+# This changes the batch size of queries in the scheduling main loop.
+# This depends on query length limits and how long you are willing to hold locks.
+# 0 for no limit
+max_tis_per_query = 0
```

Comment:
maybe add an adjective to query? max_tis_per_main_query? This sounds like it's a global to all ti queries in the scheduler.